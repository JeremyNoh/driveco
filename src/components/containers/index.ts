import TryLibContainer from './TryLibContainer';

export {TryLibContainer};
