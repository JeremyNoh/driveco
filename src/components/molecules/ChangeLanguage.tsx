import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {useTranslation} from 'react-i18next';
import {Button} from '../atoms';

export default function ChangeLanguage() {
  const {t, i18n} = useTranslation();

  const changeTraduction = () => {
    i18n.changeLanguage(i18n.language === 'en' ? 'fr' : 'en');
  };

  return (
    <View>
      <Text style={styles.title}>
        {t('INFO_APP.CURRENT_LANG')}
        {i18n.language === 'en' ? t('INFO_APP.EN') : t('INFO_APP.FR')}
      </Text>

      <Button onPress={changeTraduction} title={t('INFO_APP.CHANGE_LANG')} />
    </View>
  );
}

const styles = StyleSheet.create({
  title: {
    alignSelf: 'center',
    marginTop: 20,
    fontWeight: 'bold',
  },
});
