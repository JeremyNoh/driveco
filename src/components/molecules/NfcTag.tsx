import React, {useEffect, useState} from 'react';
import {Alert, Platform, StyleSheet, Text, View} from 'react-native';

import NfcManager, {NfcTech} from 'react-native-nfc-manager';
import {Button} from '../atoms';

export default function NfcTag() {
  const [hasNFC, setHasNFC] = useState<boolean | null>(null);
  useEffect(() => {
    const checkIsSupported = async () => {
      const deviceIsSupported = await NfcManager.isSupported();

      setHasNFC(deviceIsSupported);
      if (deviceIsSupported) {
        await NfcManager.start();
      }
    };

    Platform.OS === 'web' ? setHasNFC(false) : checkIsSupported();
  }, []);

  async function readNdef() {
    try {
      // register for the NFC tag with NDEF in it
      await NfcManager.requestTechnology(NfcTech.Ndef);
      // the resolved tag object will contain `ndefMessage` property
      const tag = await NfcManager.getTag();
      console.warn('Tag found', tag);
      Alert.alert(`Voici l'id du Badge : ${tag?.id}`);
    } catch (ex) {
      console.warn('Oops!', ex);
    } finally {
      // stop the nfc scanning
      NfcManager.cancelTechnologyRequest();
    }
  }
  if (hasNFC === null) {
    return (
      <Text testID="loadingNfcId" style={styles.text}>
        Loading
      </Text>
    );
  } else if (!hasNFC) {
    return (
      <Text testID="notAvailableNfcId" style={styles.text}>
        NFC pas Compatible
      </Text>
    );
  }
  return (
    <View testID="availableNfcId">
      <Button onPress={readNdef} title="Scan a Tag" />;
    </View>
  );
}

const styles = StyleSheet.create({
  text: {
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
