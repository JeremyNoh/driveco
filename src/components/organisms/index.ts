import TryLibOrganism from './TryLibOrganism';

export {TryLibOrganism};
