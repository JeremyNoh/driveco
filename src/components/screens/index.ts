import TryLibScreen from './TryLibScreen';

export {TryLibScreen};
