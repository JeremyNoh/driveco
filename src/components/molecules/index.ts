import NfcTag from './NfcTag';
import ChangeLanguage from './ChangeLanguage';

export {NfcTag, ChangeLanguage};
