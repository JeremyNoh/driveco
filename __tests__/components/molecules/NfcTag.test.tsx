import 'react-native';

import React from 'react';

// // Note: test renderer must be required after react-native.
import {cleanup, render, waitFor} from '@testing-library/react-native';
import {NfcTag} from '../../../src/components/molecules';

describe('NFC-Tag Molecule', () => {
  afterEach(() => cleanup());

  beforeEach(() => {
    jest.mock('react-native-nfc-manager', () => ({
      __esModule: true,
      default: {
        isSupported: jest.fn().mockReturnValue(false),
        isEnabled: jest.fn().mockReturnValue(true),
        start: jest.fn().mockResolvedValue(true),
      },
    }));
  });

  it('renders correctly', () => {
    render(<NfcTag />).toJSON();
  });
  it('Match with snapshot', () => {
    const input = render(<NfcTag />).toJSON();
    expect(input).toMatchSnapshot();
  });
  it('should show  Loading', () => {
    const {getByTestId} = render(<NfcTag />);
    expect(getByTestId('loadingNfcId').props.children).toBeDefined();
  });
  it('should show not working', async () => {
    const {getByTestId} = render(<NfcTag />);
    const wrapperForNotAvailableNFC = await waitFor(() =>
      getByTestId('notAvailableNfcId'),
    );
    expect(wrapperForNotAvailableNFC.props.children).toBeDefined();
  });
});
