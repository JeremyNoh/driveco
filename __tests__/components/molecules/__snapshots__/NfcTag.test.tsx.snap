// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFC-Tag Molecule Match with snapshot 1`] = `
<Text
  style={
    Object {
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
  testID="loadingNfcId"
>
  Loading
</Text>
`;
